GCRCatalogs/register.py

```diff
@@ -91,22 +91,39 @@ class RootDirManager:
     )
     _DICT_LIST_KEYS = ("catalogs",)
     _DESC_SITE_ENV = "DESC_GCR_SITE"
+    _NO_DEFAULT_ROOT_WARN = """
+       Default root dir has not been set; catalogs may not be found.
+       To set root dir by site 
+            $ export {}='sitename'
+       or, from within Python,
+            GCRCatalogs.set_root_dir_by_site('sitename')
+       where sitename is one of ({})
+
+       If standard site values for root dir are not appropriate, use
+            GCRCatalogs.set_root_dir('my_root_dir')
+    """
 
     def __init__(self, site_config_path=None):
         self._site_config_path = site_config_path
         self._default_root_dir = None
         self._custom_root_dir = None
+        self._site_list = []
+        self._site_config = {}
         self._site_info = self._get_site_info()
 
         if self._site_config_path:
             site_config = load_yaml_local(self._site_config_path)
+            self._site_config = site_config
+            self._site_list = list(site_config.keys())
             for k, v in site_config.items():
                 if k in self._site_info:
                     self._default_root_dir = v
                     break
 
         if not self._default_root_dir:
-            warnings.warn("Default root dir has not been set!")
+            site_string = ' '.join(self._site_list)
```

Comment:
Use public `site_list`
```suggestion
            site_string = ' '.join(self.site_list)
```